updater/updater_p3a.h

```diff
@@ -0,0 +1,49 @@
+/* Copyright (c) 2025 The Brave Authors. All rights reserved.
+ * This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this file,
+ * You can obtain one at https://mozilla.org/MPL/2.0/. */
+
+#ifndef BRAVE_UPDATER_UPDATER_P3A_H_
+#define BRAVE_UPDATER_UPDATER_P3A_H_
+
+#include <string>
+
+#include "base/time/time.h"
+#include "components/prefs/pref_registry_simple.h"
+#include "components/prefs/pref_service.h"
+
+// We're migrating Brave's auto-update mechanism to Omaha 4. As we do this, we
+// want to compare the success rates of the Omaha 4 and legacy implementations.
+// The functions in this file achieve this by reporting UMA events when the
+// browser was (or was not) updated.
+
+namespace brave_updater {
+
+inline constexpr char kUpdateStatusHistogramName[] = "Brave.Update.Status";
+
+enum UpdateStatus {
```

Comment:
`enum class`. Also adding `kMaxValue = kUpdatedWithOmaha4` should allow you to remove the third param in the UMA call